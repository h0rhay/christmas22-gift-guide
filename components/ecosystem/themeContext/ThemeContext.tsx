import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
