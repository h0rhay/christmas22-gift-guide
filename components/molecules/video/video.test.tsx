import { render } from "@testing-library/react";
import Video from ".";
import { ThemeProvider } from "styled-components";
import Theme from "styles/theme";
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

const renderComponent = (
  id: string,
  altText: string,
  imgId?: string,
  landscapeVideoId?: string,
  portraitVideoId?: string
) =>
  render(
    <ThemeProvider theme={Theme}>
      <ThemeContext colorGroup="hero">
        <Video
          id={id}
          altText={altText}
          imgId={imgId}
          landscapeVideoId={landscapeVideoId}
          portraitVideoId={portraitVideoId}
        />
      </ThemeContext>
    </ThemeProvider>
  );

describe("Video Player", () => {
  describe("Video with all properties", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_9x16"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video missing an image", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-2",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_9x16"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video when only img and landscape video are provided", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-3",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "SLF_XMAS_Guide_DeadHungry_16x9"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video when only  img and portrait video are provided", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-4",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "SLF_XMAS_Guide_DeadHungry_9x16"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video when only landscape video is provided", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-5",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video when only portrait video is provided", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-6",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_9x16"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video when no properties are provided", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-7",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video with wrong landscape ID", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-8",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "landscapeWrongId",
        "SLF_XMAS_Guide_DeadHungry_9x16"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video with wrong portrait ID", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-9",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "SLF_XMAS_Guide_DeadHungry_16x9",
        "portraitWrongId"
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Video with wrong IDs for both types", () => {
    it("should match snapshot", () => {
      const { container } = renderComponent(
        "some-id-10",
        "video-poster-image-SLF_XMAS_Guide_DeadHungry_16x9",
        "SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb",
        "landscapeWrongId",
        "portraitWrongId"
      );
      expect(container).toMatchSnapshot();
    });
  });
});
