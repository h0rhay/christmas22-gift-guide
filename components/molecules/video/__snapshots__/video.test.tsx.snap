// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Player Video missing an image should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 cKfVbs"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 hbzbcz"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video when no properties are provided should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <h2>
      No video provided
    </h2>
  </div>
</div>
`;

exports[`Video Player Video when only  img and portrait video are provided should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 cKfVbs"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 hbzbcz"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video when only img and landscape video are provided should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 cKfVbs"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 hbzbcz"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video when only landscape video is provided should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <h2>
      No video provided
    </h2>
  </div>
</div>
`;

exports[`Video Player Video when only portrait video is provided should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <h2>
      No video provided
    </h2>
  </div>
</div>
`;

exports[`Video Player Video with all properties should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 hbLzSI"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 gtaSQJ"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video with wrong IDs for both types should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 hbLzSI"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 gtaSQJ"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video with wrong landscape ID should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 hbLzSI"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 gtaSQJ"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Video Player Video with wrong portrait ID should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="video__Container-sc-16c86f4-2 hbLzSI"
      data-test-id="video-container"
    >
      <div
        class="video__Wrapper-sc-16c86f4-1 gtaSQJ"
      >
        <div
          style="opacity: 1;"
        >
          <div
            class="videoPoster__StyledVideoPoster-sc-1fts0xv-1 juXlxw"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <img
                alt="video-poster-image"
                data-nimg="fill"
                data-test-id="video-poster-img"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/SLF_XMAS_Guide_DeadHungry_16x9_Test_Thumb?qlt=80&fmt=jpg&scl=1); background-position: 0% 0%;"
              />
              <noscript />
            </div>
            <div
              class="videoPoster__StyledButton-sc-1fts0xv-0 gJXsOg"
              data-test-id="play-video"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                />
                <path
                  d="M6.271 5.055a.5.5 0 0 1 .52.038l3.5 2.5a.5.5 0 0 1 0 .814l-3.5 2.5A.5.5 0 0 1 6 10.5v-5a.5.5 0 0 1 .271-.445z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Play Video
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
